cilium/cmd/debuginfo.go

```diff
@@ -35,16 +36,21 @@ var debuginfoCmd = &cobra.Command{
 	Run:   runDebugInfo,
 }
 
-var file string
+var (
+	file string
+	html string
+)
 
 func init() {
 	rootCmd.AddCommand(debuginfoCmd)
 	debuginfoCmd.Flags().StringVarP(&file, "file", "f", "", "Redirect output to file")
+	debuginfoCmd.Flags().StringVarP(&html, "html-file", "", "", "Convert default output to HTML file")
```

Comment:
could we change this to a toggle? the interaction between html and file is confusing below, especially when we assign html to file, but then branch on len(html) but then writeHTML uses file (albeit, to read the data)! 
We'd always use file as the output path to write to, and html as a toggle for the format. This would mean having a writeHTML and writeNotHTML functions. To store the output data writeHTML uses currently, we could write to and read from a bytes.Buffer (this implements the Reader and Writer interfaces, so `Fprintf` and `printMD` would simply take it as a parameter instead of w.

Reply:
Should be as described. Let me know if I misunderstood.